build/syntax_mapping.rs

```diff
@@ -236,8 +236,14 @@ fn get_def_paths() -> anyhow::Result<Vec<PathBuf>> {
     ];
 
     let mut toml_paths = vec![];
-    for subdir in source_subdirs {
-        let wd = WalkDir::new(Path::new("src/syntax_mapping/builtins").join(subdir));
+    for subdir_name in source_subdirs {
+        let subdir = Path::new("src/syntax_mapping/builtins").join(subdir_name);
+        if !subdir.try_exists()? {
+            // Directory might not exist due to this `cargo vendor` bug:
+            // https://github.com/rust-lang/cargo/issues/15080
+            continue;
```

Comment:
Is there a way we can log a build warning here so that it is visible when the mappings are lost?

Reply:
If I understand the Cargo bug correctly from the Cargo maintainers' responses, the platform-specific mapping directory is only lost when there are no platform-specific mappings (like macos). So the person building shouldn't need to care. I don't think a warning is warranted.

Reply:
Makes sense, thanks 👍